app/models/search_suggestion.rb

```diff
@@ -2,21 +2,46 @@
 
 class SearchSuggestion
   include ApiEntity
+  CACHE_VERSION = 'v1'.freeze
 
   collection_path '/search_suggestions'
 
   attr_accessor :value
 
-  def self.cached_suggestions
-    TradeTariffFrontend::ServiceChooser.cache_with_service_choice(
-      'cached_search_suggestions', expires_in: 24.hours
-    ) do
-      all
+  class << self
+    def cached_suggestions_for(term)
+      return [] if term.blank?
+
+      first_letter = term.slice(0, 1).downcase
+      cached_suggestions_for_letter first_letter
     end
-  end
 
-  def self.start_with(term)
-    cached_suggestions.select { |s| s.value =~ /^#{term}/i }
-                      .first(TradeTariffFrontend.suggestions_count)
+    def start_with(term)
+      cached_suggestions_for(term).select { |s| s.value =~ /^#{term}/i }
+                                  .first(TradeTariffFrontend.suggestions_count)
+    end
+
+  private
+
+    def cached_suggestions_for_letter(first_letter)
+      results = Rails.cache.read_multi "#{cache_prefix}loaded",
+                                       "#{cache_prefix}#{first_letter}"
+
+      if results["#{cache_prefix}loaded"]
+        return results["#{cache_prefix}#{first_letter}"] || []
+      end
+
+      grouped_results = all.group_by { |r| r.value.to_s.downcase.slice(0, 1) }
+      grouped_results.transform_keys! { |letter| "#{cache_prefix}#{letter}" }
+      grouped_results["#{cache_prefix}loaded"] = true
+
+      Rails.cache.write_multi grouped_results, expires_in: 24.hours
```

Comment:
```suggestion
      Rails.cache.write_multi grouped_results
```

A day is the default

Reply:
Bit split on this one - elsewhere we've been explicit on the expiry, and its replacing code which was explicit about the expiry - I think I'm going to leave it as is for the moment. We do have a specific requirement for how long this is cached for since it ties in with the data sync on the backend.